cmd/kubeadm/app/cmd/config.go

```diff
@@ -138,6 +141,59 @@ func getDefaultAPIObjectBytes(apiObject string) ([]byte, error) {
 	return []byte{}, fmt.Errorf("--api-object needs to be one of %v", availableAPIObjects)
 }
 
+// NewCmdConfigMigrate returns cobra.Command for "kubeadm config migrate" command
+func NewCmdConfigMigrate(out io.Writer) *cobra.Command {
```

Comment:
What's the user story here?  I'm just wondering if adding a cli makes sense and will be used, b/c once created it will be near impossible to take out.  

especially seeing how we have diff now. 

/cc @liztio

Reply:
This is client-side migration of config files (bytes on disk) only, it doesn't touch any upgrading code nor is a replacement for anything upgrade-specific.